tests/client/sdk/users/test_api.py

```diff
@@ -12,25 +12,103 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+from uuid import uuid4
+
 import pytest
+from argilla.client.api import ArgillaSingleton
 from argilla.client.client import Argilla
 from argilla.client.sdk.client import AuthenticatedClient
-from argilla.client.sdk.commons.errors import BaseClientError, UnauthorizedApiError
-from argilla.client.sdk.users.api import whoami
-from argilla.client.sdk.users.models import User
+from argilla.client.sdk.commons.errors import (
+    AlreadyExistsApiError,
+    BaseClientError,
+    ForbiddenApiError,
+    NotFoundApiError,
+    UnauthorizedApiError,
+)
+from argilla.client.sdk.users.api import create_user, delete_user, list_users, whoami
+from argilla.client.sdk.users.models import UserModel
+from argilla.server.models import User
+
+from tests.factories import UserFactory
 
 
-def test_whoami(api: Argilla):
+def test_whoami(api: Argilla) -> None:
     user = whoami(client=api.http_client)
-    assert isinstance(user, User)
+    assert isinstance(user, UserModel)
 
 
-def test_whoami_with_auth_error(api: Argilla):
-    with pytest.raises(UnauthorizedApiError):
-        api.http_client.token = "wrong_token"
-        whoami(api.http_client)
+def test_whoami_errors() -> None:
+    with pytest.raises(
+        BaseClientError, match="Your Api endpoint at http://localhost:6900 is not available or not responding"
+    ):
+        whoami(AuthenticatedClient(base_url="http://localhost:6900", token="wrong-apikey"))
 
 
-def test_whoami_with_connection_error():
-    with pytest.raises(BaseClientError):
-        whoami(AuthenticatedClient(base_url="http://localhost:6900", token="wrong-apikey"))
+def test_list_users(owner: User) -> None:
+    UserFactory.create(username="user_1")
+    UserFactory.create(username="user_2")
+    httpx_client = ArgillaSingleton.init(api_key=owner.api_key).http_client.httpx
+
+    response = list_users(client=httpx_client)
```

Comment:
For this kind of tests I think you can use directly the `client: TestClient`, fixture
```suggestion
def test_list_users(client: TestClient, owner: User, owner_auth_header: dict) -> None:
    UserFactory.create(username="user_1")
    UserFactory.create(username="user_2")
    client.headers.update(owner_auth_header)

    response = list_users(client=client)
```

Reply:
But won't the `ArgillaSingleton` use the `TestClient` as configured in the `conftest.py`? Is this change worth it? Asking because if so, we should change this is more than one place, LMKWYT!

Reply:
Yes, just to be more explicit here in tests if we're using the HTTP client directly. The conftest is a fixture to use the rg.init functions with an already mocked client